### PR TITLE
auditing: Take daily snapshots of table history

### PR DESCRIPTION
**Pull request by underyx**

### CODE REVIEW

File: zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace: constant
```
Comment (aexvir) [inline zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml]: Shouldn't the namespace be different here?

File: zoo/auditing/tasks.py [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
     for issue_id, user_name, reverse_url in issues:
         issue = Issue.objects.get(id=issue_id)
         create_git_issue(issue, user_name, reverse_url)
+
+
+@shared_task
+def take_issue_table_snapshots():
+    issues = Issue.objects.all()
+    for row in issues.values("repository_id", "status").annotate(count=Count("*")):
+        IssueCountByRepositorySnapshot(**row).save()
```
Comment (aexvir) [inline zoo/auditing/tasks.py]: I'd call `full_clean` before saving. Unless you are using a form for this, which you aren't, doing this is not ensuring that the status is one of the choices.

File: zoo/auditing/tasks.py [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+    for row in issues.values("kind_key", "status").annotate(count=Count("*")):
+        IssueCountByKindSnapshot(**row).save()
```
Comment (aexvir) [inline zoo/auditing/tasks.py]: Same for this one

**Review by aexvir — CHANGES_REQUESTED** [review]

**Comment by aexvir** [comment]
And run coala and black locally as well

File: zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml (hunk 1 above)
Comment (underyx) [reply zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml]: yeowch

File: zoo/auditing/tasks.py (hunk 2 above)
Comment (underyx) [reply zoo/auditing/tasks.py]: eek

File: zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml (hunk 1 above)
Comment (underyx) [reply zoo/auditing/standards/dummy_standards/metadata/check_randomly.yml]: changed to `random`

File: zoo/auditing/tasks.py (hunk 3 above)
Comment (underyx) [reply zoo/auditing/tasks.py]: Same for this one

File: zoo/auditing/tasks.py [hunk 4]
```diff
@@ -9,3 +10,16 @@ def bulk_create_git_issues(issues):
...
+
+@shared_task
+def take_issue_table_snapshots():
+    issues = Issue.objects.all()
+    for row in issues.values("repository_id", "status").annotate(count=Count("*")):
+        snapshot = IssueCountByRepositorySnapshot(**row)
+        snapshot.full_clean()
+        snapshot.save()
+    for row in issues.values("kind_key", "status").annotate(count=Count("*")):
+        snapshot = IssueCountByKindSnapshot(**row).save()
```
Comment (aexvir) [inline zoo/auditing/tasks.py]: Don't save it yet

**Review by aexvir — CHANGES_REQUESTED** [review]

File: zoo/auditing/tasks.py (hunk 4 above)
Comment (underyx) [reply zoo/auditing/tasks.py]: goddamn last line effect

**Review by aexvir — APPROVED** [review]

**Pull request merged**